etna/transforms/math/sklearn.py

```diff
@@ -113,7 +116,18 @@ def fit(self, df: pd.DataFrame) -> "SklearnTransform":
         self.transformer.fit(X=x)
         return self
 
-    def transform(self, df: pd.DataFrame) -> pd.DataFrame:
+    def fit(self, ts: TSDataset) -> "SklearnTransform":
+        """Fit the transform."""
+        super().fit(ts)
+        self.in_column_regressor = [True if col in ts.regressors else False for col in self.in_column]  # type: ignore
+        self.out_column_regresors = [
```

Comment:
You need to define it inside the constructor instead of `self.in_column_regressor`